lib/beetle/dead_letter_queue.rb

```diff
@@ -0,0 +1,90 @@
+require 'net/http'
+require 'json'
+
+module Beetle
+  class DeadLetterQueue
+    READ_TIMEOUT = 3 #seconds
+    DEFAULT_DEAD_LETTER_MSG_TTL = 1000 #1 second
+    RABBIT_API_PORT = 15672
+
+    class FailedRabbitRequest < StandardError; end
+
+    class << self
+      def bind_dead_letter_queues!(channel, servers, target_queue, creation_keys = {})
+        dead_letter_queue_name = dead_letter_queue_name(target_queue)
+
+        logger.debug("Beetle: creating dead letter queue #{dead_letter_queue_name} with opts: #{creation_keys.inspect}")
+        dead_letter_queue = channel.queue(dead_letter_queue_name, creation_keys)
+
+        logger.debug("Beetle: setting #{dead_letter_queue_name} as dead letter queue of #{target_queue} on all servers")
+        set_dead_letter_queues!(servers, target_queue)
+
+        logger.debug("Beetle: setting #{target_queue} as dead letter queue of #{dead_letter_queue_name} on all servers")
+        set_dead_letter_queues!(servers, dead_letter_queue_name,
+                                :message_ttl => DeadLetterQueue::DEFAULT_DEAD_LETTER_MSG_TTL,
+                                :routing_key => target_queue)
+      end
+
+      def set_dead_letter_queues!(servers, queue_name, options={})
+        servers.each { |server| set_dead_letter_queue!(server, queue_name, options) }
+      end
+
+      def set_dead_letter_queue!(server, queue_name, options={})
+        raise ArgumentError.new("server missing")     if server.blank?
+        raise ArgumentError.new("queue name missing") if queue_name.blank?
+
+        request_url = URI("http://#{server}/api/policies/%2F/#{queue_name}_policy")
```

Comment:
We should not hardcode the vhost (/ aka %2F) here, but use the one from the config.